### PR TITLE
chore(workflow): add push to subtree reusable

### PR DESCRIPTION
**Pull request by rbioteau**
Integrated in bonita-studio-sp: https://github.com/bonitasoft/bonita-studio-sp/blob/dev/.github/workflows/push_to_subtree.yml

### CODE REVIEW

File: .github/workflows/_reusable_push_to_subtree.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          Default is 'dev'. Can't be a tag.
+    secrets:
+      BONITA_CI_PAT:
+        required: true
+
+jobs:
+  push-to-subtree:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
```
Comment (akantcheff) [inline .github/workflows/_reusable_push_to_subtree.yml:34]: No need to use our git-setup-action?

Comment (rbioteau) [reply .github/workflows/_reusable_push_to_subtree.yml:34]: No, but it is a good point, it looks like the hash of the commit is changed and the verification of the commit is lost. This is annoying... I'll try using another action using the same tool that look better (https://github.com/frankdejonge/use-subsplit-publish?tab=readme-ov-file#example-workflow)

Comment (akantcheff) [reply .github/workflows/_reusable_push_to_subtree.yml:36]: Can we do it with a classic git command instead of using a custom github action?

Comment (rbioteau) [reply .github/workflows/_reusable_push_to_subtree.yml:36]: No it is more complicated than that. It needs to download the tools...etc
I don't known what git subtree or splitsh do, but they do not make a "real" commit and it looks not possible to generate signed commits](https://github.com/splitsh/lite/issues/52)

**Pull request merged**